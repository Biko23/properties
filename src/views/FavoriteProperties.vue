<template>
<div>
  <favorite-properties-component/>
</div>
</template>

<script>
import FavoritePropertiesComponent from '../components/FavoritePropertiesComponent.vue'

export default {
  name: 'Property',

  components: {
    FavoritePropertiesComponent
  }
}
</script>
