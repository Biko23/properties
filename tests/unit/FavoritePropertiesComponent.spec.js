import { createLocalVue, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import VueRouter from 'vue-router'
import Vuex from 'vuex'
import FavoriteProperties from '@/components/FavoritePropertiesComponent.vue'

describe('FavoritePropertiesComponent.vue', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.use(VueRouter)
  let vuetify
  let getters
  let actions
  let store
  let router

  beforeEach(() => {
    vuetify = new Vuetify()
    router = new VueRouter()
    actions = {
      fetchAllDetailedCurrentUserProperties: jest.fn(),
      removePropertyFromFavoriteSection: jest.fn(),
      postAUserLog: jest.fn()
    }
    getters = {
      allDetailedCurrentFavoriteList: () => ({}),
    }
    store = new Vuex.Store({
      getters,
      actions
    })
  })


  const mountFactory = options => {
    return mount(FavoriteProperties, {
      localVue,
      vuetify,
      store,
      ...options
    })
  }

  it('should match snapshot', () => {
    const wrapper = mountFactory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
