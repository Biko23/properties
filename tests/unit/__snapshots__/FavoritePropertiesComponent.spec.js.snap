// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritePropertiesComponent.vue should match snapshot 1`] = `
<div>
  <div class="container container--fluid" id="container">
    <div id="property-header" class="row">
      <div style="flex: 1;">
        <h3>Properties</h3> <small data-testid="total-result-element" style="font-weight: bold;"> results</small>
      </div>
      <div style="flex: 1;">
        <h3 style="color: rgb(59, 110, 243);">Favorite List</h3>
      </div>
    </div>
    <div id="main-property" class="row"></div>
  </div><br>
</div>
`;
